torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4900,6 +4900,9 @@ def sample_inputs_index(op_info, device, dtype, requires_grad, **kwargs):
     copy = op_info.name == "index_copy"
     # target.index_fill(dim, idx, value)
     fill = op_info.name == "index_fill"
+    # target._index_reduce(dim, idx, source, reduce, *, include_self=True)
+    reduce_op = op_info.name == "_index_reduce"
```

Comment:
Modeling sample input functions that can produce different samples like this is probably not what you want to do. Typically you want to partial the function with new inputs that control its behavior (or if the ops require very heterogenous inputs just have different sample input functions). 

See the addmm sample input specification for an example:

```
sample_inputs_func=partial(sample_inputs_addmm, alpha=1, beta=1),
```

Reply:
Thanks, will fix in a follow up!